driver/SDL/SDL_keyboard.c

```diff
@@ -0,0 +1,150 @@
+/**
+ * @file keyboard.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+#include "SDL_keyboard.h"
+#if USE_KEYBOARD != 0
+
+/*********************
+ *      DEFINES
+ *********************/
+#ifndef MONITOR_ZOOM
+#define MONITOR_ZOOM    1
+#endif
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+static bool keyup = false;
+static bool keydown = false;
+static bool repeat = false;
+static bool keyboard_was_read = true;
+static int last_key = -1;
```

Comment:
These key state globals are accessed from different threads right? The SDL thread and Micropython thread.  
How do you ensure atomic updates of this state? (that all of the key state globals are written before being read by the other thread)